Runtime/CourseValidation/CheckForComponentAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Innoactive.Creator.Core.Configuration;
+using Innoactive.Creator.Core.SceneObjects;
+using UnityEngine;
+
+namespace Innoactive.Creator.Core.Validation
+{
+    /// <summary>
+    /// Checks if the referenced GameObject has at least one of the listed Type added as Component.
+    /// </summary>
+    public class CheckForComponentAttribute : Attribute, IAttributeValidator
+    {
+        /// <inheritdoc/>
+        public virtual ValidationErrorLevel ErrorLevel { get; }
+
+        protected string message = "One Component of {0} is missing";
+
+        protected List<Type> components;
+
+        public CheckForComponentAttribute(ValidationErrorLevel errorLevel, params Type[] components)
+        {
+            ErrorLevel = errorLevel;
+            this.components = components.ToList();
+        }
+
+        public CheckForComponentAttribute(params Type[] components) : this (ValidationErrorLevel.ERROR, components) { }
+
+        /// <inheritdoc/>
+        public virtual bool Validate(object value, out string message)
+        {
+            message = "";
+            if (IsMissingComponent(FetchGameObject(value)))
+            {
+                message = String.Format(this.message, string.Join(", ", components));
+                return true;
+            }
+
+            return false;
+        }
+
+        protected virtual bool IsMissingComponent(GameObject gameObject)
+        {
+            if (gameObject == null)
+            {
+                return false;
+            }
+
+            return components.All(component => gameObject.GetComponent(component) == null);
+        }
+
+        protected virtual GameObject FetchGameObject(object value)
+        {
+            if (value == null || !(value is UniqueNameReference reference))
```

Comment:
Please, stick to the code convention.

```suggestion
            if (value == null || (value is UniqueNameReference reference) == false)
```